src/Nethermind/Nethermind.Consensus.Ethash/EthashSealValidator.cs

```diff
@@ -90,17 +91,23 @@ public bool ValidateParams(BlockHeader parent, BlockHeader header)
             bool extraDataNotTooLong = header.ExtraData.Length <= 32;
             if (!extraDataNotTooLong)
             {
-                _logger.Warn(
-                    $"Invalid block header ({header.ToString(BlockHeader.Format.Full)}) - extra data too long {header.ExtraData.Length}");
+                if (_logger.IsWarn) _logger.Warn($"Invalid block header ({header.ToString(BlockHeader.Format.Full)}) - extra data too long {header.ExtraData.Length}");
                 return false;
             }
 
             UInt256 difficulty = _difficultyCalculator.Calculate(header, parent);
             bool isDifficultyCorrect = difficulty == header.Difficulty;
             if (!isDifficultyCorrect)
             {
-                _logger.Warn(
-                    $"Invalid block header ({header.ToString(BlockHeader.Format.Full)}) - incorrect difficulty {header.Difficulty} instead of {difficulty}");
+                if (_logger.IsWarn) _logger.Warn($"Invalid block header ({header.ToString(BlockHeader.Format.Full)}) - incorrect difficulty {header.Difficulty} instead of {difficulty}");
+                return false;
+            }
+
+            ulong unixTimeSeconds = _timestamper.UnixTime.Seconds;
```

Comment:
so we started validating blocks using clock, right?

Reply:
only in PoW